repo/fsrepo/migrations/ipfsdir.go

```diff
@@ -0,0 +1,129 @@
+package migrations
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/mitchellh/go-homedir"
+)
+
+const (
+	envIpfsPath = "IPFS_PATH"
+	defIpfsDir  = ".ipfs"
+	versionFile = "version"
+
+	// Local IPFS API
+	apiFile = "api"
+)
+
+func init() {
+	homedir.DisableCache = true
+}
+
+// ApiEndpoint reads the api file from the local ipfs install directory and
+// returns the address:port read from the file.  If the ipfs directory is not
+// specified then the default location is used.
+func ApiEndpoint(ipfsDir string) (string, error) {
```

Comment:
@gammazero this is moving out right since go-ipfs-api does this anyway and that's getting called by the relevant binaries (ipfs-update + fs-repo-migrations)?

Reply:
Yes.  That should have been moved out.

Reply:
Done.